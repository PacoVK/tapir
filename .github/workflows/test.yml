name: Test

on:
  push:

jobs:
  docker:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v3
      - name: Setup JDK
        uses: actions/setup-java@v3
        with:
          java-version: '17'
          distribution: 'adopt'
          cache: maven
      - name: Run tests
        run: ./mvnw verify
