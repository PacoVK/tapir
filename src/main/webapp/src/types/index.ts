export interface Module {
    namespace: string;
    name: string;
    provider: Provider;
}

export enum Provider {
    AWS = "aws",
    GOOGLE = "google",
    AZURE = "azurerm",
    KUBERNETES = "kubernetes"
}