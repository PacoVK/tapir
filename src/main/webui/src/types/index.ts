export interface Module {
  id: string;
  namespace: string;
  name: string;
  provider: Provider;
  downloads: number;
  published_at: string;
  versions: { version: string }[];
}

export interface Misconfiguration {
  resource: string;
  severity: string;
  rule_description: string;
  description: string;
  impact: string;
  resolution: string;
  links: string;
  location: {
    filename: string;
    start_line: number;
    end_line: number;
  };
}
export enum Provider {
  AWS = "aws",
  GOOGLE = "google",
  AZURE = "azurerm",
  KUBERNETES = "kubernetes",
}

export interface TabPanelProps {
  children?: React.ReactNode;
  index: number;
  value: number;
}

export interface ModuleInputProps {
  inputs: {
    name: string;
    description: string;
    type: string;
    required: boolean;
  }[];
}

export interface ModuleOutputProps {
  outputs: {
    name: string;
    description: string;
  }[];
}

export interface ModuleResourcesProps {
  resources: {
    name: string;
    type: string;
  }[];
}

export interface ReportFindingsProps {
  reports: { [p: string]: unknown } | ArrayLike<unknown>;
}

export interface ModuleDependenciesProps {
  modules: {
    source: string;
    version: string;
  }[];

  providers: {
    name: string;
    alias: string;
    version: string;
  }[];
}

export interface ModuleAnalysisTabProps {
  version: string;
  reports: any;
  documentation: any;
}
