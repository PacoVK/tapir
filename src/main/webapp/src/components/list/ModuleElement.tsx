import React from "react";
import {Box, Paper, Stack} from "@mui/material";
import awsLogo from "./../../assets/aws-icon.png";
import gcpLogo from "./../../assets/google-icon.png";
import azureLogo from "./../../assets/azurerm-icon.png";
import k8Logo from "./../../assets/kubernetes-icon.png";
import defaultLogo from "./../../assets/terraform-icon.png";
import {Module, Provider} from "../../types";
import Typography from "@mui/material/Typography";

const getProviderLogo = (provider: Provider) => {
    switch (provider) {
        case Provider.AWS:
            return awsLogo;
        case Provider.GOOGLE:
            return gcpLogo;
        case Provider.AZURE:
            return azureLogo;
        case Provider.KUBERNETES:
            return k8Logo;
        default:
            return defaultLogo;
    }
}

const ModuleElement = ({module}: {module: Module}) => {
    return(
        <Box marginY={2}>
            <Paper>
                <Stack direction={"row"} spacing={4}>
                    <img
                        style={{
                            marginTop: 10,
                            marginBottom: 10,
                            marginLeft: 10,
                            width: 64,
                            height: 64,
                        }}
                        alt={"Provider logo"}
                        src={getProviderLogo(module.provider)}
                    />
                    <Box>
                        <Typography variant="h5" marginY={2}>
                            {module.namespace}/{module.name}
                        </Typography>
                    </Box>
                </Stack>
                <Box style={{backgroundColor: "yellow"}}>t</Box>
            </Paper>
        </Box>
    );
}

export default ModuleElement;