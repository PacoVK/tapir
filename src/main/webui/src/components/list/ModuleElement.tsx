import React from "react";
import { Box, Chip, Paper, Stack } from "@mui/material";
import { Module } from "../../types";
import Typography from "@mui/material/Typography";
import DownloadIcon from "@mui/icons-material/Download";
import RocketLaunchIcon from "@mui/icons-material/RocketLaunch";
import InfoIcon from "@mui/icons-material/Info";
import { getProviderLogo } from "../../util/LogoUtil";
import { formatDate } from "../../util/DateUtil";
import { Link } from "react-router-dom";

const ModuleElement = ({ module }: { module: Module }) => {
  return (
    <Box
      marginBottom={2}
      component={Link}
      to={`module/${module.namespace}/${module.name}/${module.provider}`}
      state={{ data: module }}
    >
      <Paper>
        <Stack direction={"row"} spacing={4} alignItems={"center"}>
          <img
            style={{
              marginTop: 10,
              marginBottom: 10,
              marginLeft: 10,
              width: 64,
              height: 64,
            }}
            alt={"Provider logo"}
            src={getProviderLogo(module.provider)}
          />
          <Box>
            <Typography variant="h5" marginY={2}>
              {module.namespace}/{module.name}
            </Typography>
          </Box>
        </Stack>
        <Stack direction={"row"} spacing={2} ml={2}>
          <Chip
            icon={<DownloadIcon />}
            label={`Total downloads: ${module.downloads}`}
          />
          <Chip
            icon={<InfoIcon />}
            label={`Latest version: ${module.versions.at(0)!.version}`}
          />
          <Chip
            icon={<RocketLaunchIcon />}
            label={`Last published at: ${formatDate(module.published_at)}`}
          />
        </Stack>
      </Paper>
    </Box>
  );
};

export default ModuleElement;
