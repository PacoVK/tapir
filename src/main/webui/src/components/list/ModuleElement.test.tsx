import ModuleElement from "./ModuleElement";
import { ProviderType } from "../../types";
import { renderWithRouter } from "../../test/RouterUtils";

describe("<ModuleElement /> spec", () => {
  it("renders the ModuleElement", () => {
    const view = renderWithRouter(
      <ModuleElement
        module={{
          id: "foo/bar/baz",
          namespace: "foo",
          name: "bar",
          provider: ProviderType.AWS,
          downloads: 0,
          versions: [{ version: "1.0" }],
          published_at: "2021-10-10T10:10:10.000Z",
        }}
      />,
    );
    expect(view).toMatchSnapshot();
  });
});
