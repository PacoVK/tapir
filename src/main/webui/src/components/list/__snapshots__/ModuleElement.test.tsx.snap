// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModuleElement /> spec renders the ModuleElement 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="MuiBox-root css-1qm1lh"
        href="/module/foo/bar/aws"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
        >
          <div
            class="MuiStack-root css-6xevoz-MuiStack-root"
          >
            <img
              alt="Provider logo"
              src="aws-icon.png"
              style="margin-top: 10px; margin-bottom: 10px; margin-left: 10px; width: 64px; height: 64px;"
            />
            <div
              class="MuiBox-root css-0"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 css-1sj8qfn-MuiTypography-root"
              >
                foo
                /
                bar
              </h5>
            </div>
          </div>
          <div
            class="MuiStack-root css-3vpec0-MuiStack-root"
          >
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-i4bv87-MuiSvgIcon-root"
                data-testid="DownloadIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 20h14v-2H5v2zM19 9h-4V3H9v6H5l7 7 7-7z"
                />
              </svg>
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                Total downloads: 0
              </span>
            </div>
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-i4bv87-MuiSvgIcon-root"
                data-testid="InfoIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                />
              </svg>
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                Latest version: 1.0
              </span>
            </div>
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-i4bv87-MuiSvgIcon-root"
                data-testid="RocketLaunchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9.19 6.35c-2.04 2.29-3.44 5.58-3.57 5.89L2 10.69l4.05-4.05c.47-.47 1.15-.68 1.81-.55l1.33.26zM11.17 17s3.74-1.55 5.89-3.7c5.4-5.4 4.5-9.62 4.21-10.57-.95-.3-5.17-1.19-10.57 4.21C8.55 9.09 7 12.83 7 12.83L11.17 17zm6.48-2.19c-2.29 2.04-5.58 3.44-5.89 3.57L13.31 22l4.05-4.05c.47-.47.68-1.15.55-1.81l-.26-1.33zM9 18c0 .83-.34 1.58-.88 2.12C6.94 21.3 2 22 2 22s.7-4.94 1.88-6.12C4.42 15.34 5.17 15 6 15c1.66 0 3 1.34 3 3zm4-9c0-1.1.9-2 2-2s2 .9 2 2-.9 2-2 2-2-.9-2-2z"
                />
              </svg>
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                Last published at: 10-10-2021
              </span>
            </div>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="MuiBox-root css-1qm1lh"
      href="/module/foo/bar/aws"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
      >
        <div
          class="MuiStack-root css-6xevoz-MuiStack-root"
        >
          <img
            alt="Provider logo"
            src="aws-icon.png"
            style="margin-top: 10px; margin-bottom: 10px; margin-left: 10px; width: 64px; height: 64px;"
          />
          <div
            class="MuiBox-root css-0"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-1sj8qfn-MuiTypography-root"
            >
              foo
              /
              bar
            </h5>
          </div>
        </div>
        <div
          class="MuiStack-root css-3vpec0-MuiStack-root"
        >
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-i4bv87-MuiSvgIcon-root"
              data-testid="DownloadIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 20h14v-2H5v2zM19 9h-4V3H9v6H5l7 7 7-7z"
              />
            </svg>
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Total downloads: 0
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-i4bv87-MuiSvgIcon-root"
              data-testid="InfoIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
              />
            </svg>
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Latest version: 1.0
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-i4bv87-MuiSvgIcon-root"
              data-testid="RocketLaunchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9.19 6.35c-2.04 2.29-3.44 5.58-3.57 5.89L2 10.69l4.05-4.05c.47-.47 1.15-.68 1.81-.55l1.33.26zM11.17 17s3.74-1.55 5.89-3.7c5.4-5.4 4.5-9.62 4.21-10.57-.95-.3-5.17-1.19-10.57 4.21C8.55 9.09 7 12.83 7 12.83L11.17 17zm6.48-2.19c-2.29 2.04-5.58 3.44-5.89 3.57L13.31 22l4.05-4.05c.47-.47.68-1.15.55-1.81l-.26-1.33zM9 18c0 .83-.34 1.58-.88 2.12C6.94 21.3 2 22 2 22s.7-4.94 1.88-6.12C4.42 15.34 5.17 15 6 15c1.66 0 3 1.34 3 3zm4-9c0-1.1.9-2 2-2s2 .9 2 2-.9 2-2 2-2-.9-2-2z"
              />
            </svg>
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Last published at: 10-10-2021
            </span>
          </div>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
